src/replay/consensus.zig

```diff
@@ -1234,3 +1316,200 @@ test "checkAndHandleNewRoot - success" {
     }
     try testing.expect(!fixture.progress.map.contains(hash1.slot));
 }
+
+test "computeBankStats - child bank heavier" {
+    var prng = std.Random.DefaultPrng.init(91);
+    const random = prng.random();
+
+    // Set up slots and hashes for the fork tree: 0 -> 1 -> 2 -> 3
+    const root = SlotAndHash{ .slot = 0, .hash = Hash.initRandom(random) };
+    const hash1 = SlotAndHash{ .slot = 1, .hash = Hash.initRandom(random) };
+    const hash2 = SlotAndHash{ .slot = 2, .hash = Hash.initRandom(random) };
+    const hash3 = SlotAndHash{ .slot = 3, .hash = Hash.initRandom(random) };
+
+    var fixture = try TestFixture.init(testing.allocator, root);
+    defer fixture.deinit(testing.allocator);
+
+    try fixture.fill_keys(testing.allocator, random, 1);
+
+    // Create the tree of banks in a BankForks object
+    var trees1 = try std.BoundedArray(TreeNode, MAX_TEST_TREE_LEN).init(0);
+    trees1.appendSliceAssumeCapacity(&[3]TreeNode{
+        .{ hash1, root },
+        .{ hash2, hash1 },
+        .{ hash3, hash2 },
+    });
+    try fixture.fill_fork(
+        testing.allocator,
+        .{ .root = root, .data = trees1 },
+        true,
+    );
+
+    const my_node_pubkey = fixture.node_pubkeys.items[0];
+    const votes = [_]u64{2};
+    for (votes) |vote| {
+        _ = vote;
+        // const result = fixture.simulate_vote(vote, my_vote_pubkey, &fixture.tower);
+        // try testing.expectEqual(@as(usize, 0), result.len);
+    }
+
+    var frozen_slots = try fixture.slot_tracker.frozenSlots(
+        testing.allocator,
+    );
+    defer frozen_slots.deinit(testing.allocator);
+    errdefer frozen_slots.deinit(testing.allocator);
+
+    // Convert ancestors for computeBankStats
+    var ancestors = try convertAncestorsMap(testing.allocator, &fixture.ancestors);
+    defer ancestors.deinit(testing.allocator);
+
+    // TODO move this into fixture?
+    const versioned_stakes = try testEpochStakes(
+        testing.allocator,
+        fixture.vote_pubkeys.items,
+        10000,
+        random,
+    );
+    defer versioned_stakes.deinit(testing.allocator);
+
+    var epoch_stakes = std.AutoHashMapUnmanaged(Epoch, VersionedEpochStakes).empty;
+    defer epoch_stakes.deinit(testing.allocator);
+    try epoch_stakes.put(testing.allocator, 0, versioned_stakes);
+    try epoch_stakes.put(testing.allocator, 1, versioned_stakes);
+
+    var epoch_schedule = EpochSchedule.DEFAULT;
```

Comment:
does this need to be var? if not it should be const.

Reply:
[b789fe2](https://github.com/Syndica/sig/pull/812/commits/b789fe2297419c1c0adbcf8f129643977b2b9201)